src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/ComponentDatabase.java

```diff
@@ -9,8 +10,25 @@
     return component == null ? null : component.getParent();
   }
 
-  public static @Nullable Component getEditorWindow() {
-    return ComponentLocator.getComponentByClass("EditorWindow");
+  /**
+   * Returns any of the open editors - it is not deterministic which one will be returned.
+   */
+  public static @Nullable EditorComponentImpl getEditorWindow() {
```

Comment:
The result of this function is then passed to `EditorHighlighter`, which ultimately has to call public methods of `EditorComponentImpl`. Yes, this could return Component, but there must be _some_ cast along the way so that `EditorHighlighter` can use this component.

Reply:
Yeah, you're absolutely correct. Let's keep the return type as-is. :+1: